torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1441,8 +1442,184 @@ void InitXlaModuleBindings(py::module m) {
   });
 
   BuildProfilerSubmodule(&m);
-}
 
+  m.def("_get_tensors_handle",
+        [](const std::vector<at::Tensor>& tensors) -> std::vector<int64_t> {
+          std::vector<int64_t> handles;
+          handles.reserve(tensors.size());
+          for (auto& tensor : tensors) {
+            handles.push_back(bridge::GetXlaTensor(tensor)->GetOpaqueHandle());
+          }
+          return handles;
+        });
+
+  // -------------Dynamo Integration API Start-------------------------
+  /*
+   * Return tensor ids and tensors for DeviceData nodes.
+   */
+  m.def("_get_tensors_xla_device_data_node",
+        [](const std::vector<at::Tensor>& tensors)
+            -> std::pair<std::vector<int64_t>, std::vector<at::IValue>> {
+          std::vector<int64_t> tensor_ids;
+          std::vector<at::IValue> ivalues;
+          std::vector<torch::lazy::Node*> roots;
+          for (auto& tensor : tensors) {
+            auto xtensor = bridge::TryGetXlaTensor(tensor);
```

Comment:
we should check for this value before appending to the list, this might segfault if tensor is not an xlaTensor.